examples/mongocxx/get_values_from_documents.cpp

```diff
@@ -0,0 +1,127 @@
+// Copyright 2016 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+
+#include <bsoncxx/builder/stream/document.hpp>
+#include <mongocxx/instance.hpp>
+#include <mongocxx/client.hpp>
+
+using namespace bsoncxx;
+
+// Document model, showing array with nested documents:
+//
+// {
+//     "_id" : ObjectId(...),
+//     "messagelist" : [
+//       {
+//         "uid" : 413098706,
+//         "status" : 2,
+//         "msg": "..."
+//       },
+//       ...
+//     ]
+// }
+
+// Construct a document in the format of 'messagelist'
+document::value new_message(int64_t uid, int32_t status, std::string msg) {
+    builder::stream::document doc;
+    return doc << "uid" << uid << "status" << status << "msg" << msg << builder::stream::finalize;
+}
+
+// Insert a document into the database
+void insert_test_data(mongocxx::collection& coll) {
+    builder::stream::document builder{};
+    builder << "messagelist" << builder::stream::open_array
+            << new_message(413098706, 3, "Lorem ipsum...")
+            << new_message(413098707, 2, "Lorem ipsum...")
+            << new_message(413098708, 1, "Lorem ipsum...") << builder::stream::close_array;
+
+    document::value doc = builder << builder::stream::finalize;
+
+    // Normally, one should check the return value for success
+    coll.insert_one(doc.view());
+}
+
+// Iterate over contents of messagelist
+void iterate_messagelist(document::element& ele) {
+    // Check validity && type before trying to iterate
+    if (ele.type() == type::k_array) {
+        array::view subarray{ele.get_array().value};
+        for (array::element msg : subarray) {
+            // Check correct type before trying to access elements.
+            // Only print out fields if they exist; don't report missing fields.
+            if (msg.type() == type::k_document) {
+                document::view subdoc{msg.get_document().value};
+                if (subdoc["uid"] && subdoc["uid"].type() == type::k_int64) {
```

Comment:
Suggest pulling out the `subdoc["uid"]` element into a local variable (perhaps `uidEle`?), instead of calculating it a few times.  Same with the other elements in this block.